src/plugins/stdioproc/stdioproc.c

```diff
@@ -0,0 +1,576 @@
+/**
+ * @file
+ *
+ * @brief Source for stdioproc plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "stdioproc.h"
+
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbinvoke.h>
+#include <kdbprivate.h>
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+typedef struct
+{
+	pid_t childPid;
+	FILE * toChild;
+	FILE * fromChild;
+	Key * childContractKey;
+	KeySet * childContract;
+	ElektraInvokeHandle * dump;
+	struct
+	{
+		bool open;
+		bool get;
+		bool set;
+		bool close;
+	} hasOp;
+} IoData;
+
+#define MSG_HANDSHAKE_HEADER_V1 "ELEKTRA_STDIOPROC INIT v1\n"
```

Comment:
const char *?

Reply:
Then we can't use `sizeof`. In most other cases `strlen` would work and the compiler would optimize it away. But in the snippet below, it would change the meaning of the code.

https://github.com/ElektraInitiative/libelektra/blob/38d1d1789759aa8968575ad857e38f90d9c8c9ae/src/plugins/stdioproc/stdioproc.c#L148